Indicators/VolumeWeightedAveragePriceIndicator.cs

```diff
@@ -78,7 +76,7 @@ public override void Reset()
         {
             _price.Reset();
             _volume.Reset();
-            _vwap.Reset();
+            _vwap = _price.WeightedBy(_volume, _period);
```

Comment:
bug?

Reply:
Yes. Resetting `_vwap` is not enough, because `WeightedBy` has local variables that cannot be reset.